src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/inlet.hpp"
+
+#include <iostream>
+
+#include "fmt/fmt.hpp"
+
+#include "axom/slic/core/SimpleLogger.hpp"
+
+namespace inlet = axom::inlet;
+using Vector = inlet::FunctionType::Vec3D;
+
+// Used to convert an unordered map (the type Inlet uses) to represent arrays
+// to a geometric vector type
+// FIXME: Clean this up/remove when PR #429 is merged
+Vector mapToVector(const std::unordered_map<int, double>& map)
+{
+  Vector result;
+  // Make sure the elements are accessed in order
+  for(int i = 0; i < 3; i++)
+  {
+    const auto ele = map.find(i);
+    if(ele != map.end())
+    {
+      result[i] = ele->second;
+    }
+  }
+  return result;
+}
+
+// A union of the members required for each of the operations is stored for simplicity
+struct Operator
+{
+  double rotate;
+  Vector axis;
+  Vector center;
+  Vector translate;
+  double x;
+  double y;
+  double z;
+  enum class Type
+  {
+    Translate,
+    Rotate,
+    Slice
+  } type;
+
+  // Again, the union of the necessary members are defined as part of the schema
+  static void defineSchema(inlet::Table& table)
+  {
+    table.addDouble("rotate");
+    // Vectors are defined as arrays of doubles
+    table.addDoubleArray("axis");
+    table.addDoubleArray("center");
+    table.addDoubleArray("translate");
+    table.addDouble("x");
+    table.addDouble("y");
+    table.addDouble("z");
+  }
+};
+
+template <>
+struct FromInlet<Operator>
+{
+  Operator operator()(const inlet::Table& base)
+  {
+    Operator result;
+    // Even though all the possible members are part of the schema, the
+    // "contains" operation can be used to check what is actually present
+    // in the input file
+    if(base.contains("translate"))
+    {
+      result.type = Operator::Type::Translate;
+      result.translate = mapToVector(base["translate"]);
+    }
+    else if(base.contains("rotate"))
+    {
+      result.type = Operator::Type::Rotate;
+      result.rotate = base["rotate"];
+      result.axis = mapToVector(base["axis"]);
+      result.center = mapToVector(base["center"]);
+    }
+    else if(base.contains("x") || base.contains("y") || base.contains("z"))
```

Comment:
FWIW, in my example, "slice" was a nested dictionary:
```
operators:
   translate: [1, 2, 3]
   slice:
      x: 10
```
The idea is that you can slice along an x, y, or z plane. There are also other options available. For example, to tell it which direction should be "up" in the new coordinate system:
```
slice:
   x: 10
   up: [1, 2]
```

If you think this could trip up the implementation, please make sure that the tests cover situations like these.

Reply:
Having seen your example below, it looks like maybe this should be `base.contains("slice")`. Is this right, or does `contains()` also go to children?

Reply:
> "slice" was a nested dictionary

Ah, the spacing must have gotten lost in translation.  I've updated the example accordingly.  The schema adjustments mean that the conditional in `FromInlet<Operator>` can use the more intuitive `base.contains("slice")`.

Reply:
As far as I can see, this should not affect @joshessman-llnl's solution.

Perhaps the operator could have `slice_x`, `slice_y`, ... `slice_up`, if that would be helpful/clearer.